--- a/recognition/include/pcl/recognition/face_detection/face_detector_data_provider.h	2026-08-24 01:10:42.289652235 +0800
+++ b/recognition/include/pcl/recognition/face_detection/face_detector_data_provider.h	2026-08-24 01:10:42.305673702 +0800
@@ -12,7 +12,7 @@
 #include <pcl/recognition/face_detection/face_common.h>
 
 #include <boost/algorithm/string.hpp>
-#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem.hpp>
 
 #include <fstream>
 #include <string>
